src/cmd/linuxkit/registry/manifest.go

```diff
@@ -11,15 +11,23 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-var platforms = []string{
-	"linux/amd64", "linux/arm64", "linux/s390x",
+// these platforms are used only as the source for registry.PushManifestList(). Since it is
+// configured to ignore missing, we could include dozens if we wanted; it doesn't hurt, adding maybe a few seconds to
+// the whole run.
+// Ideally, we could just look for all tags that start with linuxkit/foo:<hash>-*, but the registry API
+// only supports "list all the tags" and "get a specific tag", no "get by pattern". The "get a specific tag"
+// is exactly what registry.PushManifestList() uses, so no benefit to use doing that in advance,
+// while "list all tags" is slow, and has to cycle through all of the (growing numbers of) tags
+// before we know what exists. Might as well leave it as is.
+var platformsToSearchForIndex = []string{
+	"linux/amd64", "linux/arm64", "linux/s390x", "linux/riscv64", "linux/386", "linux/ppc64le", "linux/arm/v6", "linux/arm/v7",
```

Comment:
Lets not put the 32 bit platforms, we have never built them and I don't want to support 32 bit. We will have enough fun with armv9 and stuff no doubt...

This is kind of horrible, we should pull the existing manifest and add/replace the architectures really.

Reply:
> This is kind of horrible, we should pull the existing manifest and add/replace the architectures really.

It isn't that simple. In principle, we are not doing, "I have added the following local arch, so let me add that arch," it is a more generic, "tell manifest-tool to add every arch I know about, skipping the ones I don't know." 

If hub had a "get tags that match x-*", then I would get those and add them. But that doesn't exist, only get a specific one or list all (see the comment right above this line).

If someone pushes out just the arch-specific on platform A, then on B, then C, then D, and then runs the manifest, it should create for all of them. I don't know of any way to do it other than to list every arch we could think of.

Reply:
Removed the 32-bit ones.

Reply:
Yeah, there is something wrong with the whole manifest list design...

Reply:
I think the issue actually is with the registry. If I could do queries, like "get tags by starts with x" or "by architecture" or "by annotation", then many more things become possible and flexible.

Reply:
I don't think this should be the role of tags. Ideally you would want to query by tags (which images and architectures are built from this git commit?) or equivalent.

Reply:
That would be fine too. But I don't think any of that exists.